packages/gatsby/src/services/start-webpack-server.ts

```diff
@@ -109,9 +112,45 @@ export async function startWebpackServer({
         webpackActivity.end()
         webpackActivity = null
       }
-      enqueueFlush()
+
+      if (isSuccessful) {
+        const state = store.getState()
+        const mapOfTemplatesToStaticQueryHashes = mapTemplatesToStaticQueryHashes(
+          state,
+          stats.compilation
+        )
+
+        mapOfTemplatesToStaticQueryHashes.forEach(
+          (staticQueryHashes, componentPath) => {
+            if (
+              !isEqual(
+                state.staticQueriesByTemplate.get(componentPath)?.sort(),
+                staticQueryHashes.map(String)?.sort()
```

Comment:
I think in these situations where you know a value will exist you should use the bang instead of a conditional. The `?` actually adds runtime code whereas the `!` just tells TS that a property _will_ exist

Reply:
TIL, nice! Thank you!

Reply:
Shouldn't be needed at all though. `Array.prototype.map` isn't nullable. The `map.get` arguably does need a runtime check, so you should probably keep the optional chain there.

Reply:
Good point, done